[CCM-17060] : Cluster orchestrator documentation (#6903)
harness/developer-hub: docs/cloud-cost-management/4-use-ccm-cost-optimization/3-Cluster-orchestrator/1-introduction.md

## AWS Spot Instances and challenges 
Amazon EC2 provides a range of purchase options catering to diverse computing needs, including On-Demand Instances, Reserved Instances, and Spot Instances. While On-Demand Instances offer flexibility with pay-as-you-go pricing, Reserved Instances provide significant cost savings with long-term commitments. Spot Instances, on the other hand, allow users to bid for unused EC2 capacity at lower prices, making them ideal for fault-tolerant, flexible and stateless workloads. However, Spot Instances come with challenges such as potential interruptions and the risk of termination within just 2 minutes of a notice if capacity becomes unavailable.

<DocImage path={require('./static/AWS_pricing.png')} width="70%" height="70%" title="Click to view full size image" />

## Cluster Infrastructure and challenges  

<DocImage path={require('./static/challenges.png')} width="50%" height="50%" title="Click to view full size image" />

#### Balancing Workload Requirements

> Matching workload demands with cloud infrastructure can be challenging due to their evolving nature. Specifically, autoscaling cluster nodes to meet workload needs presents difficulties as workload requirements change over time.

Reviewer: ```suggestion
Because workload demands are always evolving, matching these demands with cloud infrastructure can be challenging. Specifically, autoscaling cluster nodes to meet workload needs presents difficulties because workload requirements change over time.
```

#### Pod Autoscaling
Kubernetes offers pod autoscaling capabilities, including horizontal pod autoscaling (HPA) and vertical pod autoscaling (VPA). HPA adjusts pod counts based on workload demand fluctuations, while VPA dynamically adjusts pod resource allocations. However, seamlessly matching these changes with appropriate nodes presents complexities. Incorrect node selection may result in pending pods due to insufficient resources or scaling down the wrong nodes, leading to potential disruptions in workload deployment.

## Harness Cluster Orchestrator for Amazon Elastic Kubernetes Service (EKS)

The Harness Cluster Orchestrator for Amazon Elastic Kubernetes Service (EKS), a component of the Harness Cloud Cost Management (CCM) module, is designed to tackle the obstacles listed above. It intelligently scales EKS cluster nodes according to actual workload requirements. Additionally, by leveraging CCM’s distributed Spot orchestration capability, you can save up to 90% on cloud costs with Amazon EC2 Spot Instances.

> It also facilitates workload-driven autoscaling, dynamically adjusting cluster resources to match workload demands while optimizing the utilization of AWS Spot instances. This approach ensures efficient resource allocation and cost-effective cluster management. Harness’  Cluster Orchestrator for Amazon EKS distinguishes itself from other solutions in the market because of additional features like: distributed spot orchestration, cloud autostopping, integration with commitment orchestrator, and cloud cost governance. 

Author: resolved
